### PR TITLE
feat: Change timestamp field in transactions

### PR DESCRIPTION
**Pull request by gfyrag**
# Change timestamp field in transactions from string to time.Time

This change the type of the field 'Timestamp' inside transactions.

Both sqlite and Postgres can handle encoding of the field without additional efforts.
But, at least sqlite, if the db column is not defined as a date type, is not able to decode the value directly to a time.Time struct.
So the decoding is still manual for reading transactions (ok on logs since the column has a date type).

Both SQLite and Postgres encode time.Time as rfc3339nano.

So, to keep backward compatibility, we :
* Round the dates to the second internally (on transactions, and logs)
* Add a custom marshal function on transactions to format dates as rfc3339

We could change the precision in a later release.

Also the PR set the same date for the log, and the generated transactions.
It was not the case previously. I don't think it is a breaking change since the logs is not exposed anywhere actually.

The PR also remove verbose logging on tests because the tests give so many line and the CI refuse to display all the lines.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Refactoring / Technical debt

## What parts of the code are impacted ?
* pkg/core
* pkg/storage

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/workflows/pr_open.yml [hunk 1]
```diff
@@ -66,7 +66,7 @@ jobs:
           name: control-dist
           path: cmd/control/
       - name: run tests
-        run: go test -tags json1 -v -coverpkg ./... -coverprofile coverage.out -covermode atomic ./...
```
Comment (jdupas22) [inline .github/workflows/pr_open.yml]: Can you add why/what this is in the PR description please ?

File: pkg/storage/sqlstorage/log.go [hunk 2]
```diff
@@ -91,7 +91,6 @@ func (s *Store) AppendLog(ctx context.Context, logs ...core.Log) error {
 func (s *Store) lastLog(ctx context.Context, exec executor) (*core.Log, error) {
 	var (
 		l    core.Log
-		ts   sql.NullString
```
Comment (jdupas22) [inline pkg/storage/sqlstorage/log.go:94]: why is this removed ? shouldnt it have been replaced by a NullTime ?

**Review by jdupas22 — DISMISSED** [review]

File: pkg/storage/sqlstorage/log.go (hunk 2 above)
Comment (gfyrag) [reply pkg/storage/sqlstorage/log.go:94]: The Scan() method is called directly with a pointer on the date : 
```err := rows.Scan(&l.ID, &l.Type, &l.Hash, &l.Date, &data)```
No need to handle NULL values as the date is required.

File: .github/workflows/pr_open.yml (hunk 1 above)
Comment (gfyrag) [reply .github/workflows/pr_open.yml]: I removed that because the tests give so many lines than the CI does not show all and we don't see errors sometimes.

File: pkg/ledger/process.go [hunk 3]
```diff
@@ -99,14 +99,16 @@ func (l *Ledger) processTx(ctx context.Context, ts []core.TransactionData) (*Com
...
 
 		tx := core.Transaction{
-			TransactionData:   t,
-			ID:                nextTxId,
-			Timestamp:         time.Now().UTC().Format(time.RFC3339),
+			TransactionData: t,
+			ID:              nextTxId,
+			// Until v1.5.0, dates was stored as string using rfc3339 format
+			// So round the date to the second to keep the same behaviour
+			Timestamp:         time.Now().UTC().Round(time.Second),
```
Comment (antoinegelloz) [inline pkg/ledger/process.go]: I think that `Truncate` instead of `Round` would be more correct to mimic former behavior

File: pkg/core/log.go [hunk 4]
```diff
@@ -107,7 +109,7 @@ func NewSetMetadataLog(previousLog *Log, metadata SetMetadata) Log {
 	l := Log{
 		ID:   id,
 		Type: SetMetadataType,
-		Date: time.Now().UTC(),
+		Date: time.Now().UTC().Round(time.Second),
```
Comment (antoinegelloz) [inline pkg/core/log.go]: Why do you need to round here?

Comment (gfyrag) [reply pkg/core/log.go]: To keep consistent behavior between all dates in the system.

File: pkg/ledger/process.go (hunk 3 above)
Comment (gfyrag) [reply pkg/ledger/process.go]: You're right

**Review by jdupas22 — DISMISSED** [review]

**Review by jdupas22 — DISMISSED** [review]

**Review by flemzord — APPROVED** [review]
